skimage/util/process.py

```diff
@@ -0,0 +1,89 @@
+from math import ceil
+from multiprocessing import cpu_count
+
+import dask.array as da
+
+__all__ = ['process_chunks']
+
+
+def _get_chunks(shape, ncpu):
+    """
+    Split the array into equal sized chunks based on the number of
+    available processors. The last chunk in each dimension absorbs the
+    remainder array elements if the number of cpus does not divide evenly into
+    the number of array elements.
+
+    >>> _get_chunks((4, 4), 4)
+    ((2, 2), (2, 2))
+    >>> _get_chunks((4, 4), 2)
+    ((2, 2), (4,))
+    >>> _get_chunks((5, 5), 2)
+    ((2, 3), (5,))
+    >>> _get_chunks((2, 4), 2)
+    ((1, 1), (4,))
```

Comment:
Ideally we would split the larger dimensions more aggressively than the smaller.  I agree with your sentiment though that we should probably wait before optimizing this much further; this is a potential rabbit hole.

In dask.array I'm honestly fine requiring users to think about chunk sizes.  Automatic solutions in n-dimensions are likely to fail decently often and users should probably be made somewhat aware of what's going on.  So far people also seem willing to think about chunk sizes; it doesn't seem to be that alien to people yet.